web/src/search_suggestion.ts

```diff
@@ -26,11 +26,15 @@ type TermPattern = Omit<NarrowTerm, "operand"> & Partial<Pick<NarrowTerm, "opera
 
 type Suggestion = {
     description_html: string;
-    search_string: string;
     is_person?: boolean;
     user_pill_context?: UserPillItem;
 };
 
+type SuggestionPart = {
+    suggestion_parts: Suggestion[];
+    search_string: string;
+};
```

Comment:
I'm a little confused by the terminology here; a `SuggestionPart` containing several `suggestion_parts` seems a bit backwards :)

A possibly clearer way to write this would be to do the opposite naming scheme:
```
type SuggestionPart = {
    description_html: string;
    is_person?: boolean;
    user_pill_context?: UserPillItem;
};

type Suggestion = {
    suggestion_parts: SuggestionPart[];
    search_string: string;
};
```

I think that might involve less churn on the existing code, in addition to having more natural naming?

Reply:
Yeah I was a bit confused too what would be proper terminology here.